gcp_variant_transforms/libs/preprocess_reporter.py

```diff
@@ -59,12 +64,16 @@ def generate_conflicts_report(header_definitions,
   inferred_headers = inferred_headers or VcfHeader()
   content_lines = []
   content_lines.extend(_generate_conflicting_headers_lines(
-      _extract_conflicts(header_definitions.formats), resolved_headers.formats))
+      _extract_conflicts(header_definitions.infos), resolved_headers.infos,
+      vcf_header_io.HeaderTypeConstants.INFO))
   content_lines.extend(_generate_conflicting_headers_lines(
-      _extract_conflicts(header_definitions.infos), resolved_headers.infos))
-  content_lines.extend(_generate_inferred_headers_lines(inferred_headers.infos))
-  content_lines.extend(_generate_inferred_headers_lines(
-      inferred_headers.formats))
+      _extract_conflicts(header_definitions.formats), resolved_headers.formats,
+      vcf_header_io.HeaderTypeConstants.FORMAT))
+  if not isinstance(inferred_headers, pvalue.EmptySideInput):
```

Comment:
This check looks weird to me. 
1) Why do you need this check? What happen if empty obj is passed to `_generate_inferred_headers_lines`? Should it take care of it?

2) This is kind of ugly because we mentioned type of `inferred_headers` as VcfHeader, but we are checking for another type here.

Reply:
Yes, I think there is an option in beam that you can tell replace empty side input with default value.